ios-target-app/Onboarding/Views/SignUpViewController.swift

```diff
@@ -8,22 +8,167 @@
+    )
+    
+    private lazy var nameField = UITextField(target: self)
+    
+    private lazy var emailLabel = UILabel(
+        text: "email_label".localized,
+        size: .normal
+    )
+    
+    private lazy var emailField = UITextField(target: self)
+    
+    private lazy var passwordLabel = UILabel(
+        text: "password_label".localized,
+        size: .normal
+    )
+    
+    private lazy var passwordField = UITextField(
+        target: self,
+        placeholder: "sign_up_password_placeholder".localized,
+        isPassword: true
+    )
+    
+    private lazy var confirmPasswordLabel = UILabel(
+        text: "sign_up_confirm_password_label".localized,
+        size: .normal
+    )
+    
+    private lazy var confirmPasswordField = UITextField(
+        target: self,
+        isPassword: true
+    )
+    
+    private lazy var genderLabel = UILabel(
+        text: "sign_up_gender_label".localized,
+        size: .normal
+    )
+    
+    private lazy var genderField = UITextField(
+        target: self,
+        placeholder: "sign_up_gender_placeholder".localized
+    )
+    
+    private lazy var genderPickerView: UIPickerView = {
+        let picker = UIPickerView()
+        picker.translatesAutoresizingMaskIntoConstraints = false
+        return picker
+    }()
+    
+    private lazy var signUpButton = UIButton(text: "sign_up_button".localized)
+    
+    private lazy var lineView = UIView()
+        
+    private lazy var signInButton = UIButton(
+        color: .clear,
+        text: "sign_in_button".localized,
+        textColor: .black,
+        target: self,
+        action: #selector(signInTapped)
+    )
+    
+    private lazy var scrollView: UIScrollView = {
+        let scrollView = UIScrollView()
+        scrollView.translatesAutoresizingMaskIntoConstraints = false
+        return scrollView
+    }()
+        
+    private lazy var stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.distribution = .equalSpacing
+        stackView.spacing = 15
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        return stackView
+    }()
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        
+        setViews()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+      super.viewWillAppear(animated)
+      navigationController?.setNavigationBarHidden(false, animated: true)
+    }
+    
+    private func setViews() {
+        view.addSubviews([
+            ovalsImageView,
+            scrollView
+        ])
+        view.backgroundColor = .white
+        scrollView.addSubview(stackView)
+        
+        let ovalsImage = UIImage(named: "ovalsGroup")
+        ovalsImageView.contentMode = UIView.ContentMode.scaleAspectFill
+        ovalsImageView.frame.size.width = view.bounds.width
+        ovalsImageView.frame.size.height = UI.OvalImageView.height
+        ovalsImageView.image = ovalsImage
```

Comment:
same comment of the previous vc. Also i'd make a heper/component for this to avoid code repetition